lib/backend/k8s/k8s_test.go

```diff
@@ -1967,6 +1987,227 @@ var _ = testutils.E2eDatastoreDescribe("Test Watch support", testutils.Datastore
+			event := ExpectAddedEvent(watch.ResultChan())
+			Expect(event.New.Key.String()).To(Equal("NetworkPolicy(default/test-net-policy-3)"))
+		})
+		It("supports watching all networkpolicies", func() {
+			watch, err := c.Watch(ctx, model.ResourceListOptions{Kind: apiv3.KindNetworkPolicy}, "")
+			Expect(err).NotTo(HaveOccurred())
+			ExpectAddedEvent(watch.ResultChan())
+		})
+	})
+
+	Describe("watching Custom Resources", func() {
+		createTestIPPool := func(name string) {
+			cidr := "192.168.0.0/16"
+			pool := &model.KVPair{
+				Key: model.ResourceKey{
+					Name: name,
+					Kind: apiv3.KindIPPool,
+				},
+				Value: &apiv3.IPPool{
+					TypeMeta: metav1.TypeMeta{
+						Kind:       apiv3.KindIPPool,
+						APIVersion: apiv3.GroupVersionCurrent,
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: name,
+					},
+					Spec: apiv3.IPPoolSpec{
+						CIDR: cidr,
+					},
+				},
+			}
+			_, err := c.Create(ctx, pool)
+			Expect(err).NotTo(HaveOccurred())
+		}
+		deleteAllIPPools := func() {
+			err := c.Clean()
+			Expect(err).NotTo(HaveOccurred())
+		}
+		BeforeEach(func() {
+			createTestIPPool("test-ippool-1")
+			createTestIPPool("test-ippool-2")
+		})
+		AfterEach(func() {
+			deleteAllIPPools()
+		})
+		It("supports watching a specific custom resource (IPPool)", func() {
+			watch, err := c.Watch(ctx, model.ResourceListOptions{Name: "test-ippool-1", Kind: apiv3.KindIPPool}, "")
+			Expect(err).NotTo(HaveOccurred())
+			event := ExpectAddedEvent(watch.ResultChan())
+			Expect(event.New.Key.String()).To(Equal("IPPool(test-ippool-1)"))
+		})
+		It("supports watching all custom resources (IPPool)", func() {
+			watch, err := c.Watch(ctx, model.ResourceListOptions{Kind: apiv3.KindIPPool}, "")
+			Expect(err).NotTo(HaveOccurred())
+			ExpectAddedEvent(watch.ResultChan())
+		})
+	})
+
+	Describe("watching Custom WorkloadEndpoints", func() {
+		createTestPod := func(name string) {
+			pod := &k8sapi.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      name,
+					Namespace: "default",
+				},
+				Spec: k8sapi.PodSpec{
+					NodeName: "127.0.0.1",
+					Containers: []k8sapi.Container{
+						{
+							Name:    "container1",
+							Image:   "busybox",
+							Command: []string{"sleep", "3600"},
+						},
+					},
+				},
+			}
+			_, err := c.ClientSet.CoreV1().Pods("default").Create(pod)
+			Expect(err).NotTo(HaveOccurred())
+		}
+		deleteAllPods := func() {
+			var zero int64
+			err := c.ClientSet.CoreV1().Pods("default").DeleteCollection(&metav1.DeleteOptions{GracePeriodSeconds: &zero}, metav1.ListOptions{})
+			Expect(err).NotTo(HaveOccurred())
+		}
+		BeforeEach(func() {
+			createTestPod("test-pod-1")
+			createTestPod("test-pod-2")
+		})
+		AfterEach(func() {
+			deleteAllPods()
+		})
+		It("supports watching a specific workloadEndpoint", func() {
+			watch, err := c.Watch(ctx, model.ResourceListOptions{Name: "default/127.0.0.1-k8s-test--pod--1-eth0)", Kind: apiv3.KindWorkloadEndpoint}, "")
```

Comment:
Ah, in this case `default/` is not part of the pod name - it's the pod's namespace.